lib/alertmanager_api/silence.py

```diff
@@ -0,0 +1,185 @@
+            "endsAt": silence.end_time_str,
+            "createdBy": silence.author,
+            "comment": silence.comment,
+        }
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silences"
+            response = requests.post(
+                api_url,
+                auth=self.auth,
+                headers={"Accept": "application/json"},
+                json=payload,
+                timeout=10,
+            )
+            if response.status_code != 200:
+                msg = (
+                    "Failed to create silence. "
+                    f"Response status code: {response.status_code} "
+                    f"Response text: {response.text}"
+                )
+                self.log.error(msg)
+                raise RequestException(msg, response=response)
+            silence_id = response.json()["silenceID"]
+            silence.silence_id = silence_id
+            return silence
+
+        except RequestException as req_ex:
+            msg = f"Failed to create silence in Alertmanager: {req_ex} "
+            if req_ex.response is not None:
+                msg += (
+                    f"Response status code: {req_ex.response.status_code} "
+                    f"Response text: {req_ex.response.text}"
+                )
+            self.log.critical(msg)
+            raise req_ex
+
+    def remove_silence(self, silence):
+        """
+        Removes a previously scheduled silence in alertmanager
+
+        :param silence: object including the silence_id to be removed
+        :type silence: SilenceDetails
+        :raises MissingMandatoryParamError: when the silence_id is missing
+        :raises requests.exceptions.RequestException:
+            when the request to the AlertManager failed
+        """
+        if not silence.silence_id:
+            msg = "Missing silence silence_id"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silence/{silence.silence_id}"
+            response = requests.delete(
+                api_url,
+                auth=self.auth,
+                timeout=10,
+            )
+            if response.status_code != 200:
+                msg = (
+                    f"Failed to delete silence with ID {silence.silence_id}. "
+                    f"Response status code: {response.status_code} "
+                    f"Response text: {response.text}"
+                )
+                self.log.critical(msg)
+                raise RequestException(msg, response=response)
+        except RequestException as req_ex:
+            msg = f"Failed to delete silence in Alertmanager: {req_ex} "
+            if req_ex.response is not None:
+                msg += (
+                    f"Response status code: {req_ex.response.status_code} "
+                    f"Response text: {req_ex.response.text}"
+                )
+            self.log.critical(msg)
+            raise req_ex
```

Comment:
Would something like this work here and also for schedule_downtime above?
```suggestion
            response.raise_for_status()
        except HTTPError as req_ex:
            msg = f"Failed to delete silence in Alertmanager: {req_ex} "
            if req_ex.response is not None:
                msg += (
                    f"Failed to delete silence with ID {silence.silence_id}. "
                    f"Response status code: {response.status_code} "
                    f"Response text: {response.text}"
                )
            self.log.critical(msg)
            raise req_ex
```